xbmc/cores/VideoPlayer/Process/ProcessInfo.cpp

```diff
@@ -600,3 +600,24 @@ int64_t CProcessInfo::GetMaxTime()
   CSingleLock lock(m_stateSection);
   return m_timeMax;
 }
+
+//******************************************************************************
+// settings
+//******************************************************************************
+CVideoSettings CProcessInfo::GetVideoSettings()
+{
+  CSingleLock lock(m_settingsSection);
+  return m_videoSettings;
+}
+
+CVideoSettings& CProcessInfo::UpdateVideoSettigs()
+{
+  CSingleLock lock(m_settingsSection);
+  return m_videoSettings;
```

Comment:
`m_videoSettings` shouldn't be accessed after lock is destructed

Reply:
I know but that needs an additional round to get fixed. currently there is no lock at all

Reply:
update is threadsafe now